code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm

```diff
@@ -209,27 +210,29 @@
 	var/spiderling_number = 1
 	var/list/enemies = list()
 
-/obj/structure/spider/eggcluster/terror_eggcluster/New()
-	..()
+/obj/structure/spider/eggcluster/terror_eggcluster/Initialize(mapload)
+	. = ..()
 	GLOB.ts_egg_list += src
-	spawn(50)
-		if(spiderling_type == /mob/living/simple_animal/hostile/poison/terror_spider/red)
+
+/obj/structure/spider/eggcluster/terror_eggcluster/proc/update_name()
```

Comment:
Moved to its own proc to kill off a `spawn(50)`. Will await response from @Kyep to see if this was done in a super specific way, but from what I can see, the change on 191 should fix any issues this has.